### PR TITLE
Change snack bar default hitTestBehavior to deferToChild when SnackBarThemeData.insetPadding is not null

### PR DESCRIPTION
**Pull request by hany-achraf**
The PR changes the default value of hitTestBehavior in snack bars to `HitTestBehavior.deferToChild` when snackBarTheme.insetPadding is not null, so that widgets behind snack bars affected by the value set to insetPadding, remain interactive even while a snack bar is visible. This PR can be considered as an extension to what have been done in PR #127959 which fixes the same problem but for individual snack bars with margin not being null. This PR works on the theme level.

*List which issues are fixed by this PR. You must list at least one issue. An issue is not required if the PR fixes something trivial like a typo.*
#148566

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] I followed the [breaking change policy] and added [Data Driven Fixes] where supported.
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[Data Driven Fixes]: https://github.com/flutter/flutter/wiki/Data-driven-Fixes

### CODE REVIEW

File: packages/flutter/lib/src/material/snack_bar.dart [hunk 1]
```diff
@@ -387,7 +387,7 @@ class SnackBar extends StatefulWidget {
 
   /// Defines how the snack bar area, including margin, will behave during hit testing.
   ///
-  /// If this property is null and [margin] is not null, then [HitTestBehavior.deferToChild] is used by default.
+  /// If this property is null, and [margin] is not null or [snackBarTheme.insetPadding] is not null, then [HitTestBehavior.deferToChild] is used by default.
```
Comment (Piinks) [inline packages/flutter/lib/src/material/snack_bar.dart:390]: It looks like this is what is causing the doc reference failure. Can you take a look?

**Review by Piinks — COMMENTED** [review]
Hi @hany-achraf welcome!

File: packages/flutter/lib/src/material/snack_bar.dart (hunk 1 above)
Comment (hany-achraf) [reply packages/flutter/lib/src/material/snack_bar.dart]: Hi @Piinks! Thanks for flagging that issue. I've taken a look and applied the fix. All checks have now passed successfully. Please go ahead and verify that everything is working as expected on your end.

File: packages/flutter/test/material/snack_bar_test.dart [hunk 2]
```diff
@@ -4045,6 +4045,127 @@ testWidgets('SnackBarAction backgroundColor works as a Color', (WidgetTester tes
...
+    await tester.pumpAndSettle();
+
+    expect(find.text(snackbarContent), findsOneWidget);
+    await tester.tapAt(tester.getTopLeft(find.text(buttonText2)));
+    expect(find.text(snackbarContent), findsOneWidget);
+
+    expect(completer.isCompleted, true);
+  });
+
+  testWidgets('Cannot interact with widgets behind SnackBar if hitTestBehavior is set to HitTestBehavior.opaque even if insetPadding is set in SnackBarThemeData', (WidgetTester tester) async {
```
Comment (Piinks) [inline packages/flutter/test/material/snack_bar_test.dart:4109]: This passes without this change, is that expected?

Comment (hany-achraf) [reply packages/flutter/test/material/snack_bar_test.dart]: Hi @Piinks! I've just removed that test in my latest commit on this PR. Please review and let me know if you'd like me to revert that change.

**Review by Piinks — APPROVED** [review]
Thanks for the update! This LGTM!
I will need to find a second reviewer, please hold. :)

**Review by MitchellGoodwin — APPROVED** [review]
LGTM! Thank you for putting this together!

**Comment by hany-achraf** [comment]
Thank you @Piinks and @MitchellGoodwin so much!

**Pull request merged**